Mathlib/RingTheory/TwoSidedIdeal.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import Mathlib.RingTheory.Congruence
+import Mathlib.Algebra.Module.LinearMap.Basic
+import Mathlib.Tactic.Abel
+
+/-!
+# Two Sided Ideals
+
+In this file, for any `Ring R`, we reinterpret `I : RingCon R` as a two sided ideal of a ring.
```

Comment:
A short discussion about the relation with subbimodules (which we have elsewhere, but are more heavyweight) would be good to have here.

It's perhaps also worth noting something about how `RingCon S` is more general in some sense as it works for semirings, but that in practice no one cares.

Reply:
hmm. If I understand it correctly, I need subtraction when proving that `RingCon S` is `SetLike`, so it means that `RingCon S` as two-sided-Ideals won't work for semirings I think.

Reply:
Added a short description